benchmark-service/benchmark/benchmark.py

```diff
@@ -100,6 +140,29 @@ async def index(request, userdata):  # pylint: disable=unused-argument
     return await render_template('benchmark', request, userdata, 'index.html', context)
 
 
+@router.get('/compare')
+@web_authenticated_developers_only(redirect=False)
+async def compare(request, userdata):  # pylint: disable=unused-argument
+    file1 = request.query.get('file1')
+    file2 = request.query.get('file2')
+    metric = request.query.get('metrics')
+    if file1 is None or file2 is None:
+        benchmarks_context1 = None
+        benchmarks_context2 = None
+        comparisons = None
+    else:
+        benchmarks_context1 = get_benchmarks(file1)
```

Comment:
What happens if you put a garbage file name in here? Do you get a 500 or an error message?

Reply:
error message